git_machete/cmd.py

```diff
@@ -1479,6 +1479,8 @@ def discover_tree():
         roots = ["master"]
     elif "develop" in local_branches():
         roots = ["develop"]
+    else:
```

Comment:
I think we can just revert to pre-2.15.5 approach in this specific case. 2.15.4 would just assign an empty list `[]` to `roots`. Roots will be discovered on the fly, see lines ~1530 below:

```python
            debug("discover_tree()", "inferred no upstream for %s, attaching %s as a new root\n" % (b, b))
            roots += [b]
```

Reply:
So no need to raise an exception

Reply:
That seems like the best solution indeed, didn't notice that the roots are added later so it is ok to start with an empty list.